src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,147 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        if (isEmpty()) {
+            head = new Node<>(null, value, null);
+            tail = head;
+        } else {
+            tail.nextElement = new Node<>(tail, value, null);
+            tail = tail.nextElement;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+            return;
+        }
+        checkIndex(index);
+        Node<T> nextNode = searchByIndex(index);
+        Node<T> previousNode = nextNode.previousElement;
+        Node<T> newNode = new Node<>(previousNode, value, nextNode);
+        nextNode.previousElement = newNode;
+        if (previousNode == null) {
+            head = newNode;
+        } else {
+            previousNode.nextElement = newNode;
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T element: list) {
+            add(element);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return searchByIndex(index).element;
```

Comment:
we already have a checkIndex() method in searchByIndex(), so you can remove checkIndex() from get() and set() and remove() methods